docs/templates/pallet.adoc

```diff
@@ -0,0 +1,40 @@
+:source-highlighter: highlight.js
+:highlightjs-languages: rust
+:github-icon: pass:[<svg class="icon"><use href="#github-icon"/></svg>]
+= Pallet Name
+
+== Purpose
+
+This is a freeform description of the tasks that this pallet fulfills
+
+== Glossary
```

Comment:
not a blocker for this PR, just a discussion item **for the future**:
- should we have a page dedicated to glossary? I think @4meta5 mentioned this in the meeting. I like the idea.
- so that, each term in the documentation can have a link to the corresponding item in the glossary

I think encountering a glossary field in the beginning would not encourage the reader. But, we do need a glossary for sure. So, having another page dedicated to glossary, with hyperlink connections, can be a better solution imo

Reply:
If you guys agree with the above, we can create an issue about it, and have it in the backlog

Reply:
opened an issue for tracking #52